console/v1/types_console_external_log_links.go

```diff
@@ -0,0 +1,40 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleExternalLogLink is an extension for customizing OpenShift web console pod log links.
+type ConsoleExternalLogLink struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleExternalLogLinkSpec `json:"spec"`
+}
+
+// ConsoleExternalLogLinkSpec is the desired console pod log link configuration.
+type ConsoleExternalLogLinkSpec struct {
+	// text is the display text for the link
+	Text string `json:"text"`
+	// hrefTemplate is an absolute secure URL (must use https) for the log link including
+	// variables to be replaced. Supported variables, which must exactly match, include:
+	// - resourceName - name of the resource which containes the logs
+	// - resourceUID - UID of the resource which contains the logs
+	//               - e.g. `11111111-2222-3333-4444-555555555555`
+	// - containerName - name of the resource's container that contains the logs
+	// - resourceNamespace - namespace of the resource that contains the logs
+	// - podLabels - JSON representation of labels matching the pod with the logs
+	//             - e.g. `{"key1":"value1","key2":"value2"}`
+	//
+	// Resource might be a pod, build, buildconfig, deployment or deploymentconfig.
+	//
+	// e.g., https://example.com/logs?resourceName=${resourceName}&containerName=${containerName}&resourceNamespace=${resourceNamespace}&podLabels=${podLabels}
+	HrefTemplate string `json:"hrefTemplate"`
+	// namespaceFilter is a regular expression used to restrict a log link to a
+	// matching namespace (e.g., ^openshift-). If not specified, links will be
```

Comment:
`...to a matching set of namespaces...`, it may be a set of one, or many.

Reply:
I think the example has to be:
  `/^openshift-/g`
If you just provide `^openshift-`, our code isn't going to treat that like a regex 

https://github.com/openshift/console/pull/1892/files#r304608136

Reply:
@benjaminapetersen @jhadvig I don't think this comment is right... It will be treated a string:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search

> A regular expression object. If a non-RegExp object obj is passed, it is implicitly converted to a RegExp by using new RegExp(obj).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp

Reply:
My mistake! We can revise that.